src/app/modules/bounce/bounce.service.ts

```diff
@@ -110,26 +116,68 @@ const logCriticalBounce = (
   bounceDoc: IBounceSchema,
   submissionId: string | undefined,
   bounceInfo: IBounceNotification['bounce'] | undefined,
+  autoEmailRecipients: string[],
 ): void => {
-  if (bounceDoc.isCriticalBounce()) {
+  logger.warn({
+    message: 'Critical bounce',
+    meta: {
+      action: 'updateBounces',
+      hasAutoEmailed: bounceDoc.hasEmailed,
+      formId: String(bounceDoc.formId),
+      submissionId: submissionId,
+      recipients: bounceDoc.getEmails(),
+      autoEmailRecipients,
+      // We know for sure that critical bounces can only happen because of bounce
+      // notifications, so we don't expect this to be undefined
+      bounceInfo: bounceInfo,
+    },
+  })
+}
+
+const computeValidEmails = (
+  populatedForm: IPopulatedForm,
+  bounceDoc: IBounceSchema,
+): string[] => {
+  const collabEmails = populatedForm.permissionList
+    ? populatedForm.permissionList.map((collab) => collab.email)
+    : []
+  const possibleEmails = collabEmails.concat(populatedForm.admin.email)
+  return difference(possibleEmails, bounceDoc.getEmails())
```

Comment:
To clarify, what is the reason for excluding the emails in bounceDoc; currently the emails will remain there for bounceLifeSpan even if the issue has been resolved (e.g. admin has cleared inbox). We could potentially increases our chances (however slim) of contacting someone if we send to these emails as well?

Reply:
This automated email is being sent immediately after we receive the bounce notification. It's quite unlikely that the email recipients would have cleared their emails in the sliver of time between us receiving the bounce notification and sending this automated notification.

Reply:
Basically we shouldn't increase our bounce rate by sending emails which we know will bounce